lib/puppet/functions.rb

```diff
@@ -46,6 +46,14 @@
 # specifies the method that should be called for that signature. When a
 # matching signature is found, the corresponding method is called.
 #
+# Special dispatches designed to create error messages for an argument mismatch
+# can be added using the keyword `argument_mismatch` instead of `dispatch`. The
+# function appointed by an `argument_mismatch` will be called with arguments
+# just like a normal `dispatch` function would, but the function must produce a
```

Comment:
strike "function" in "function would", and change "but that function must" to "but the method must..."